src/go/rpk/pkg/cli/topic/describe_storage.go

```diff
@@ -0,0 +1,258 @@
+						r.CloudStatus.MetadataUpdatePending,
+					}
+					// If isRrr is true, LastManifestSync should be there, this
+					// nil check is possibly redundant, but we want to avoid a
+					// nil pointer dereference.
+					if isRrr && r.CloudStatus.LastManifestSync != nil {
+						row = append(row, humanReadable(*r.CloudStatus.LastManifestSync, human, "time"))
+					}
+					tw.Print(row...)
+				}
+			})
+
+			header("LOG-SEGMENT-COUNT", count, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD", "LOCAL")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogSegmentCount, r.CloudStatus.LocalLogSegmentCount)
+				}
+			})
+
+			header("LOG-OFFSET", offset, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD-START", "CLOUD-LAST", "LOCAL-START", "LOCAL-LAST")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogStartOffset, r.CloudStatus.CloudLogLastOffset, r.CloudStatus.LocalLogStartOffset, r.CloudStatus.LocalLogLastOffset)
+				}
+			})
+		},
+	}
+	p.InstallAdminFlags(cmd)
+
+	cmd.Flags().BoolVarP(&all, "print-all", "a", false, "Print all cloud storage status")
+	cmd.Flags().BoolVarP(&summary, "print-summary", "s", false, "Print the summary section")
+	cmd.Flags().BoolVar(&size, "print-size", false, "Print the log size section")
+	cmd.Flags().BoolVar(&sync, "print-sync", false, "Print the sync section")
+	cmd.Flags().BoolVar(&count, "print-count", false, "Print the log count section")
+	cmd.Flags().BoolVar(&offset, "print-offset", false, "Print the offset section")
+	cmd.Flags().BoolVarP(&human, "human-readable", "H", false, "Print the times and bytes in a human-readable form")
+
+	return cmd
+}
+
+func getLastUpload(st []Status) int {
+	recent := st[0].CloudStatus.LastManifestUpload
+	for _, s := range st {
+		if s.CloudStatus.LastManifestUpload < recent {
+			recent = s.CloudStatus.LastManifestUpload
+		}
+		if s.CloudStatus.LastSegmentUpload < recent {
+			recent = s.CloudStatus.LastSegmentUpload
+		}
+	}
+	return recent
+}
+
+func humanReadable(v int, h bool, dataType string) any {
+	if !h {
+		return v
+	}
+	switch dataType {
+	case "size":
+		return units.HumanSize(float64(v))
+	case "time":
+		return units.HumanDuration(time.Duration(v)*time.Millisecond) + " ago"
+	default:
+		panic("using human readable with an unsupported type")
+	}
+}
+
+const helpDescribeStorage = `Describe the topic storage status.
+
+This commands prints detailed information about the cloud storage status of a 
+given topic, the information is divided in 5 sections:
+
+SUMMARY
+  - Name: The topic name.
+  - Partitions: Number of partitions.
+  - Cloud Storage Mode: The partition's cloud storage mode (one of: disabled, 
+    write_only, read_only, full and read_replica).
+  - Last Upload: Delta in milliseconds since the last upload of either the 
+    partition manifest or a segment.
+
+LOG SIZE
```

Comment:
Threw up some proposals in slack--I think we can _either_ merge size and segments _or_ split by local & cloud -- basically collapse three sections into two.

I lean merging size and segments, wondering what you think.

Also I think SYNC should come either first or last, basically grouping the size & offsets since they're closely related to each other.